src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSpark.java

```diff
@@ -0,0 +1,328 @@
+                while (iterator.hasNext() && size < fixedChunkSize) {
+                    Tuple2<ShortRead, ShortRead> pair = iterator.next();
+                    list.add(pair);
+                    size += pair._1.getBases().length;
+                    size += pair._2.getBases().length;
+                }
+                return list;
+            }
+        };
+    }
+
+    /**
+     * Like Guava's {@link Iterators#transform(Iterator, com.google.common.base.Function)}, but runs a fixed number
+     * ({@code numThreads}) of transformations in parallel, while maintaining ordering of the output iterator.
+     * This is useful if the transformations are CPU intensive.
+     */
+    static <F, T> Iterator<T> transformParallel(final Iterator<F> fromIterator, final Function<F, T> function, int numThreads) {
+        if (numThreads == 1) { // defer to Guava for single-threaded case
+            return Iterators.transform(fromIterator, new com.google.common.base.Function<F, T>() {
+                @Nullable
+                @Override
+                public T apply(@Nullable F input) {
+                    return function.apply(input);
+                }
+            });
+        }
+        // use an executor service for the multi-threaded case
+        final ExecutorService executorService = Executors.newFixedThreadPool(numThreads);
+        final Queue<Future<T>> futures = new LinkedList<>();
+        return new AbstractIterator<T>() {
+            @Override
+            protected T computeNext() {
+                try {
+                    while (fromIterator.hasNext()) {
+                        if (futures.size() == numThreads) {
+                            return futures.remove().get();
+                        }
+                        F next = fromIterator.next();
+                        Future<T> future = executorService.submit(() -> function.apply(next));
+                        futures.add(future);
+                    }
+                    if (!futures.isEmpty()) {
+                        return futures.remove().get();
+                    }
+                    executorService.shutdown();
+                    return endOfData();
+                } catch (InterruptedException | ExecutionException e) {
+                    throw new GATKException("Problem running task", e);
+                }
+            }
+        };
+    }
+
+    private static void reportThreadTimings() {
+        long bwaTime = 0;
+        for (Stopwatch stopwatch : stopwatches) {
+            bwaTime += stopwatch.elapsed(TimeUnit.MILLISECONDS);
+        }
+        log.info("Time in bwa: " + (bwaTime / 1000) + "s");
+    }
+
+    // From CreateSequenceDictionary
+
+    public int NUM_SEQUENCES = Integer.MAX_VALUE;
+
+    SAMSequenceDictionary makeSequenceDictionary(final File referenceFile) {
+        return makeSequenceDictionary(referenceFile.toPath());
+    }
+
+    SAMSequenceDictionary makeSequenceDictionary(final Path referenceFile) {
```

Comment:
@droazen just told me how to do it - use `new IndexedFastaSequenceFile(file).getSequenceDictionary()` (it will return null if not there and they you can use `CreateSequenceDictionary.makeSequenceDictionary`. Let's not have that copied code here.